### ISSUE TITLE
file appender fails with EEXIST

### ISSUE THREAD

**Issue by ddombrowsky** [log4js-node/log4js-node#1189]
Why is this trying to create a directory that already exists?

The code is simple:

```
log4js.configure({
            appenders: {
                logs: {
                    type: 'file',
                    filename: '/tmp/ucp-index.log',
                },
            },
            categories: {
                index: { appenders: ['logs'], level: 'debug' },
                default: { appenders: ['logs'], level: 'debug' },
            },
        });
        logger = log4js.getLogger('index');
```

Throws this error every time:

```
Error: EEXIST: file already exists, mkdir '/tmp'
    at Object.fs.mkdirSync (fs.js:872:3)
    at Object.module.exports.makeDirSync (sys/node_modules/streamroller/node_modules/fs-extra/lib/mkdirs/make-dir.js:23:13)
    at RollingFileStream._renewWriteStream (sys/node_modules/streamroller/lib/RollingFileWriteStream.js:241:8)
    at new RollingFileWriteStream (sys/node_modules/streamroller/lib/RollingFileWriteStream.js:76:10)
    at new RollingFileStream (sys/node_modules/streamroller/lib/RollingFileStream.js:18:5)
    at openTheStream (sys/node_modules/log4js/lib/appenders/file.js:17:18)
    at fileAppender (sys/node_modules/log4js/lib/appenders/file.js:59:16)
    at Object.configure (sys/node_modules/log4js/lib/appenders/file.js:117:10)
    at clustering.onlyOnMaster (sys/node_modules/log4js/lib/appenders/index.js:89:27)
    at Object.onlyOnMaster (sys/node_modules/log4js/lib/clustering.js:85:50)
```

This didn't happen yesterday, then I ran `npm i` and now `streamroller` is updated and things are breaking.

This is `log4js` 6.4.1.

**Comment by ddombrowsky** [comment]
This is running within `electron`, and it appears this makes a difference.  Running the above test in straight node works fine.  Looks like something is hijacking `fs`.

**Comment by ddombrowsky** [comment]
It does run if I pin the version to 6.3.0 in `package.json`.

**Comment by peteriman** [comment]
What is the version of your `Electron` and its bundled `Node.js`?

**Comment by peteriman** [comment]
1. What is the version of both the `Electron` and its bundled `Node.js`?
2. What is the version of the `Node.js` that you tested straight with without issues?

**Comment by peteriman** [comment]
I think it's due to the `Node.js` version.

`fs-extra` uses the `recursive` option in `fs.makeDir` to overcome the `EEXIST` error.
https://github.com/jprichardson/node-fs-extra/blob/7edcb16a06e041826af3303f961866bf3b243dae/lib/mkdirs/make-dir.js#L25

The `recursive` option is only present from `Node.js` >=`10.12.0`

**Comment by ddombrowsky** [comment]
@peteriman 
Yes, I'm running into old version problems, it looks like.
My node is current, but electron is old, and I can't really upgrade because it would require a fairly large effort.

Electron:
`Electron v3.0.13 Chromium v66.0.3359.181 Node v10.2.0v8 v6.6.346.32`

node:
`v16.13.0`

**Comment by peteriman** [comment]
Ah, your `electron` is running NodeJS `10.2.0`.
I'll try to add some form of backward compatibility so that NodeJS `^10.0.0` is all covered for the next release.

**Comment by peteriman** [comment]
Fixed in `streamroller@3.0.3`.
Fixed in `log4js@6.4.2` (which includes `streamroller@3.0.4`).

**Comment by peteriman** [comment]
Fixed in `streamroller@3.0.3`.
Fixed in `log4js@6.4.2` (which includes `streamroller@3.0.4`).

@ddombrowsky Please try it out and get back to me. Thanks!
